salt/fileserver/roots.py

```diff
@@ -138,10 +138,16 @@ def serve_file(load, fnd):
     return ret
 
 
-def update():
+def update(**kwargs):
     """
     When we are asked to update (regular interval) lets reap the cache
     """
+    # __pub_user responsible for the runner can be passed but we don't do anything with it
+    if "__pub_user" in kwargs:
+        del kwargs["__pub_user"]
```

Comment:
Does it make sense to manually maintain a list of args that a caller passes just to delete them?

Reply:
The issue that caused the bug was the propagation of the user responsible for the runner (see https://github.com/saltstack/salt/pull/64038) and this user is passed to the fileserver backend [here](https://github.com/openSUSE/salt/blob/107de57586f0b0f784771543b942dfb6bb70453a/salt/runners/fileserver.py#L353). So I assumed that in the future other keyword arguments may reach this point even though they are not useful. 

I wanted to have at least some control of what arguments are being passed so the user doesn't just add an argument expecting some behaviour with no effect. Of course there might be a better solution :)